### ISSUE TITLE
play button disappears in Rmd files

### ISSUE THREAD

**Issue by yonicd** [rstudio/rstudio#6220]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.3.828
    OS Version      : macOS Mojave 10.14.5
    R Version       : 3.6.1 (2019-07-05)

### Steps to reproduce the problem

Open a default Rmd file in the IDE via File->New File->R Markdown (HTML)

above the first chunk insert 

![Screen Shot 2020-02-13 at 11 33 30 AM](https://user-images.githubusercontent.com/1755487/74456347-af179880-4e54-11ea-811d-e6539bcbf4e8.png)

### Describe the problem in detail

That code will knit correctly and return  

````markdown
```{r, echo = FALSE}
knitr::kable(head(iris),format = 'latex',booktabs = TRUE)
```
````

in the body of the output.

The problem is that in the interactive file the play button in the subsequent chunks disappear

and if you try to run the code inline it compiles but with an side effect error message

```r
> ```{r cars}
Error: attempt to use zero-length variable name
> summary(cars)
     speed           dist       
 Min.   : 4.0   Min.   :  2.00  
 1st Qu.:12.0   1st Qu.: 26.00  
 Median :15.0   Median : 36.00  
 Mean   :15.4   Mean   : 42.98  
 3rd Qu.:19.0   3rd Qu.: 56.00  
 Max.   :25.0   Max.   :120.00  
> ```
Error: attempt to use zero-length variable name
> 
```

### Describe the behavior you expected

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [ ] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@yonicd Thank you for raising the issue! I can reproduce this in RStudio Desktop 1.2.5033 and 1.3.829 on MacOS 10.15.4 and RStudio Server 1.3.799 on Red Hat 7.8. We'll review it as we continue development of RStudio.

A few notes:
1) This is also an issue in RStudio 1.2, so this is not a regression.
2) The buttons disappear as soon as the code is added.
3) The buttons return if another ` ```` ` is included at the end of the chunk. However, then when the file is knitted, an extra ` ```` ` appears after the correctly rendered code:
<img width="498" alt="extra_ticks" src="https://user-images.githubusercontent.com/31009952/74562580-26265d00-4f39-11ea-9bb3-f94709cda32c.png">
So either way there's an issue.

**Comment by sedaerdem** [comment]
Has anyone found out a solution to this problem? Thanks

**Comment by YoYo-Pete** [comment]
I am still experiencing this situation when working with SQL blocks in interactive workbooks.  I do not know how to work around this issue.

Mac OS Big Sur 11.0.1 (20B29)
R Studio Version 1.4.1013

![image](https://user-images.githubusercontent.com/3034349/101786947-87bd4680-3acc-11eb-9623-9df887d738bb.png)

**Comment by YoYo-Pete** [comment]
I can create a new "R Notebook" file via the File-New menu in R Studio.

I can copy and paste the chunks from the notebook with the issue into a new notebook and the play button appears as expected.

So something must happen with the actual notebook file.

Also.. it's weird we cant attach our workbook files directly here, so here is a zip of the bad file.     

I opened this file in a text editor and re-saved it and opened the new file in R Studio, but the run button still was not there. 

Making a new file and copy and pasting the SQL block in will show the play button.  Which makes me feel it is somehow related to the file, but as it's just text, it makes me thing that maybe something with history or session or something in the IDE is the real issue.

New Session and Restarting R had no effect.

[Broken_SQL.Rmd.zip](https://github.com/rstudio/rstudio/files/5673600/Broken_SQL.Rmd.zip)

**Comment by sedaerdem** [comment]
Making a new file and copy/paste did not seem to work for me. Interesting thing is that the same file works fine in my desktop, but not on my laptop. Quite annoying...

**Comment by kevinushey** [comment]
Some questions:

- When exactly does the run button disappear? Is there a specific action that causes this?
- What if you mutate the chunk with the missing run button, e.g. by removing the `}` and then adding `}` back from the chunk header. Does the run button re-appear?
- What about other mutations of the document; does anything bring the run button back?

Ultimately, we're more likely to be able to fix the issue if we can reproduce it locally.

**Comment by sedaerdem** [comment]
Hi Kevin
As soon as I open the rmd file, it does not show the play buttons on my laptop. However, they appear on my desktop. I use the latest version of R and RStudio. 

Unfortunately, the run button does not appear when I remove }

Any mutations of the document on my laptop do not change the case.

**Comment by kevinushey** [comment]
What version of RStudio are you using in each case?

**Comment by sedaerdem** [comment]
Laptop (where the issue is):  Version 1.3.1093

Desktop: 1.2.5042

Both use R 4.0.3

**Comment by sedaerdem** [comment]
Got it! I rolled back to the older version on the laptop, and it all works now. Somehow new version of RStudio did create the problem.

**Comment by sedaerdem** [comment]
@YoYo-Pete, Rstudio version 1.2.5042 sorted my issue.

**Comment by Marwa-Abdelkader** [comment]
> @yonicd Thank you for raising the issue! I can reproduce this in RStudio Desktop 1.2.5033 and 1.3.829 on MacOS 10.15.4 and RStudio Server 1.3.799 on Red Hat 7.8. We'll review it as we continue development of RStudio.
> 
> A few notes:
> 
> 1. This is also an issue in RStudio 1.2, so this is not a regression.
> 2. The buttons disappear as soon as the code is added.
> 3. The buttons return if another `` ```` `` is included at the end of the chunk. However, then when the file is knitted, an extra `` ```` `` appears after the correctly rendered code:
> 
> <img alt="extra_ticks" width="498" src="https://user-images.githubusercontent.com/31009952/74562580-26265d00-4f39-11ea-9bb3-f94709cda32c.png">
> 
> So either way there's an issue.

Yes, you're right! when I added this it worked; however I just discovered that after one of the outputs that was shown inside my rmd , this ``` was there which I haven't seen before and don't know from where they came, anyways, on removing this all the following chunks have worked properly showing the run and settings buttons without the need to add this ``` after each chunk!

**Comment by Lextuga007** [comment]
I lose the play button too on when I create new SQL chunks or if I rename the output.var - ```{sql, output.var="new-name" } ``` I haven't found a way to get the play button back so I'm relying on old scripts that haven't been affected (I recently updated R Studio to Version 1.4.1103)

**Comment by panporter** [comment]
Don't know if it applies to the original issue, but here is a solution for the disappearing play button for SQL (and presumably other non-R) chunks. (e.g. @YoYo-Pete's [problem](https://github.com/rstudio/rstudio/issues/6220#issuecomment-742566761))

@momozinho [pointed to the right direction](https://github.com/Hemken/Statamarkdown/issues/15#issuecomment-812453323): it has to do with the chunk_output_type. There seems to be a bug.


Steps to reproduce:

1. Create a simple RMD file with one SQL chunk. E.g.:
` ```{sql, connection=Conn}  `
` SELECT 1 `
` ``` `

2. Switch "Global Options - R Markdown - Show output inline for all R Markdown documents" OFF.

![image](https://user-images.githubusercontent.com/68369549/125411090-1d93b980-e3ad-11eb-8bf6-53bcacfbc75b.png)

3. Run SQL chunk.

Results in error in console:

`Error: unexpected string constant in "SELECT 1"`

4. Reload file: The play button disappears.

![image](https://user-images.githubusercontent.com/68369549/125411239-3dc37880-e3ad-11eb-9949-e4477b5ff4f1.png)


5. Switch "Global Options - R Markdown - Show output inline for all R Markdown documents" back ON.
And reload file.

![image](https://user-images.githubusercontent.com/68369549/125411027-094fbc80-e3ad-11eb-9518-719930237887.png)

Play button reappears. SQL chunk runs as expected.
    

My setting:
RStudio
Version 1.4.1717
Linux and Windows

**Comment by kevinushey** [comment]
Thanks @panporter, this helps. Here's a stand-alone document to play with:

`````
---
title: "Untitled"
output: html_document
---

```{r}
library(DBI)
conn <- dbConnect(RSQLite::SQLite(), ":memory:")
```


```{sql, connection=conn}
SELECT 1
```
`````

In my case, if I toggle Show output inline for all R Markdown documents off, then the play button disappears for me. It appears this behavior is intentional based on the code:

https://github.com/rstudio/rstudio/blob/74197252b16016bbbf4fc7da6bb18ea03e2c54c3/src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetChunks.java#L313-L328

because "execute code in console" implies that only R and Python code can be executed, since the console only supports those execution modes.

**Comment by kevinushey** [comment]
tl;dr: if you want executable SQL chunks, you need to enable inline code output, at least for now.

**Comment by mikebessuille** [comment]
The SQL behaviour is by design but the escaped chunk issue described in the original problem is still something we need to address.

**Comment by panporter** [comment]
Sad, but thank you for clarification.

I additionally consider the non-updating GUI a bug:

After toggling "Show output inline for all R Markdown documents" to OFF the play button is _still visible_, but pressing it executes the (SQL) chunk as R code not as SQL.

And: Could this behaviour be documented or shown somewhere visible?

Loosing the ability to execute chunks by switching the output method does not seem to be intuitive to me (and to beginners maybe even less).

**Comment by peperomia** [comment]
I believe that backslash before [i] is the reason why Python or R chunks in RMd lost the Run buttons in my case. When I removed the backslash the buttons appeared, and disappeared again when I put them back in (see underlined and arrows in screenshot). The removal of \ didn't affect the knitted output.
![image](https://user-images.githubusercontent.com/9896177/126878673-c48ff6bc-a7cb-4a0b-a5f1-36a4deee3f73.png)

**Comment by kevinushey** [comment]
Reproducible example:

`````
---
title: "Untitled"
output: html_document
---

\[
```{r}

```
`````

The issue occurs if you try editing after inserting `\[`. See video for example.

https://user-images.githubusercontent.com/1976582/127029746-36cac908-70a1-42bf-8f4e-d5327bbfc7d9.mov

**Comment by DarioBoh** [comment]
> tl;dr: if you want executable SQL chunks, you need to enable inline code output, at least for now.

This is still the case on Posit Cloud version `2022.12.0.353.20`. For example adding 

```
editor_options:
  chunk_output_type: console
```
it triggers the `Error: unexpected symbol` message.

**Comment by PierreENOlivier** [comment]
Hi, I am currently experiencing this when the previous chunk contains Roxygen2 comments.
